From a462b0b70932e935bcf677eb68a3c59e69e878af Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Krzysztof=20Drobi=C5=84ski?= <krzysztof@kd-solutions.pl>
Date: Wed, 8 Apr 2026 10:18:41 +0200
Subject: [PATCH] imx93: Add support for OPTEE
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

OPTEE-OS starts correctly when "opteed" is enabled for Secure Payload
Dispatcher in TF-A (tested on OP-TEE version: 4.9.0), however imx93
devices require a patch for OPTEE targets because binman does not see
the tee.bin file when it is available.

Enable conditional OPTEE support for imx93 devices.

Upstream-Status: Submitted https://lists.denx.de/pipermail/u-boot/2026-April/614838.html

Signed-off-by: Krzysztof Drobiński <krzysztof@kd-solutions.pl>
---
 arch/arm/dts/imx93-u-boot.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/dts/imx93-u-boot.dtsi b/arch/arm/dts/imx93-u-boot.dtsi
index dc86746ac90..60648bd10a5 100644
--- a/arch/arm/dts/imx93-u-boot.dtsi
+++ b/arch/arm/dts/imx93-u-boot.dtsi
@@ -69,6 +69,9 @@
 				container;
 				image0 = "a55", "bl31.bin", "0x204E0000";
 				image1 = "a55", "u-boot.bin", "0x80200000";
+#if defined(CONFIG_OPTEE)
+				image2 = "a55", "tee.bin", "0x96000000";
+#endif
 			};
 		};
 	};
-- 
2.51.0

